doc/devel/contributing.rst

```diff
@@ -0,0 +1,422 @@
+repository <https://github.com/matplotlib/matplotlib/>`__ on GitHub,
+then submit a "pull request" (PR):
+
+ 1. `Create an account <https://github.com/join>`_ on
+    GitHub if you do not already have one.
+
+ 2. Fork the `project repository
+    <https://github.com/matplotlib/matplotlib>`__: click on the 'Fork' button
+    near the top of the page. This creates a copy of the code under your
+    account on the GitHub server.
+
+ 3. Clone this copy to your local disk::
+
+        $ git clone git@github.com:YourLogin/matplotlib.git
+
+ 4. Create a branch to hold your changes::
+
+        $ git checkout -b my-feature
+
+    and start making changes. Never work in the ``master`` branch!
+
+ 5. Work on this copy, on your computer, using Git to do the version
+    control. When you're done editing, do::
+
+        $ git add modified_files
+        $ git commit
+
+    to record your changes in Git, then push them to GitHub with::
+
+        $ git push -u origin my-feature
+
+Finally, go to the web page of the your fork of the matplotlib repo,
+and click 'Pull request' to send your changes to the maintainers for review.
+You may want to consider sending an email to the mailing list for more
+visibility.
+
+.. seealso::
+   
+  * `Git documentation <https://git-scm.com/documentation>`_
+  * :ref:`development-workflow`.
+  * :ref:`using-git`
+
+Contributing pull requests
+--------------------------
+
+It is recommended to check that your contribution complies with the following
+rules before submitting a pull request:
+
+  * If your pull request addresses an issue, please use the title todescribe
+    the issue and mention the issue number in the pull request description
+    to ensure a link is created to the original issue.
+
+  * All public methods should have informative docstrings with sample
+    usage presented as doctests when appropriate. Use the
+    `numpy docstring standard <https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt>`_
+
+  * Formatting should follow `PEP8 recommandation
+    <http://www.python.org/dev/peps/pep-0008/>`_. You should consider
+    installing/enabling automatic PEP8 checking in your editor.  Part of the
+    test suite is checking PEP8 compliance, things go smoother if the code is
+    mostly PEP8 compliant to begin with.
+
+  * Each high-level plotting function should have a simple example in
+    the `Example` section of the docstring.  This should be as simple as
+    possible to demonstrate the method.  More complex examples should go
+    in the `examples` tree.
+
+  * Changes (both new features and bugfixes) should be tested. See
+    :ref:`testing` for more details.
+
+  * Import the following modules using the standard scipy conventions::
+
+      import numpy as np
+      import numpy.ma as ma
+      import matplotlib as mpl
+      from matplotlib import pyplot as plt
+      import matplotlib.cbook as cbook
+      import matplotlib.collections as mcol
```

Comment:
There's only one example that uses `mcol`, and most of the rest of the code (no examples, though) uses `mcollections`.